nifi-nar-bundles/nifi-mqtt-bundle/nifi-mqtt-processors/src/main/java/org/apache/nifi/processors/mqtt/SubscribeMQTT.java

```diff
@@ -0,0 +1,301 @@
+        final List<PropertyDescriptor> descriptors = getAbstractPropertyDescriptors();
+        descriptors.add(PROP_TOPIC_FILTER);
+        descriptors.add(PROP_QOS);
+        descriptors.add(PROP_MAX_QUEUE_SIZE);
+
+        this.descriptors = Collections.unmodifiableList(descriptors);
+
+        final Set<Relationship> relationships = new HashSet<Relationship>();
+        relationships.add(REL_MESSAGE);
+        this.relationships = Collections.unmodifiableSet(relationships);
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return this.relationships;
+    }
+
+    @Override
+    public final List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return descriptors;
+    }
+
+    @OnScheduled
+    public void onScheduled(final ProcessContext context) {
+        try {
+            maxQueueSize = context.getProperty(PROP_MAX_QUEUE_SIZE).asLong();
+
+            broker = context.getProperty(PROP_BROKER_URI).getValue();
+            String clientID = context.getProperty(PROP_CLIENTID).getValue();
+
+            connOpts = new MqttConnectOptions();
+            connOpts.setCleanSession(true);
+
+            PropertyValue sslProp = context.getProperty(PROP_SSL_CONTEXT_SERVICE);
+            if (sslProp.isSet()) {
+                Properties sslProps = transformSSLContextService((SSLContextService) sslProp.asControllerService());
+                connOpts.setSSLProperties(sslProps);
+            }
+
+            PropertyValue lastWillTopicProp = context.getProperty(PROP_LAST_WILL_TOPIC);
+            if (lastWillTopicProp.isSet()){
+                PropertyValue lastWillMessage = context.getProperty(PROP_LAST_WILL_TOPIC);
+                PropertyValue lastWillRetain = context.getProperty(PROP_LAST_WILL_TOPIC);
+                connOpts.setWill(lastWillTopicProp.getValue(), lastWillMessage.getValue().getBytes(), LAST_WILL_QOS, lastWillRetain.isSet() ? lastWillRetain.asBoolean() : false);
+            }
+
+            PropertyValue usernameProp = context.getProperty(PROP_USERNAME);
+            if(usernameProp.isSet()) {
+                connOpts.setUserName(usernameProp.getValue());
+                connOpts.setPassword(context.getProperty(PROP_PASSWORD).getValue().toCharArray());
+            }
+
+            String topicFilter = context.getProperty(PROP_TOPIC_FILTER).getValue();
+            synchronized (mqttClientLock) {
+                mqttClient = new MqttClient(broker, clientID, persistence);
+
+                mqttClient.setCallback(new SubscribeMQTTCallback());
+                getLogger().info("Connecting to broker: " + broker);
+                mqttClient.connect(connOpts);
+                mqttClient.subscribe(topicFilter, context.getProperty(PROP_QOS).asInteger());
+            }
+            autoReconnectExecutor = new ScheduledThreadPoolExecutor(1);
+        } catch(MqttException me) {
+            getLogger().error("msg "+me.getMessage());
+        }
+    }
```

Comment:
As I stated above, the logic appears to be somewhat duplicated with the other processor. Perhaps some refactoring and moving all or part of it in the abstract class.